siac/hostcmd.go

```diff
@@ -45,8 +49,137 @@ You may also supply a specific address to be announced, e.g.:
 Doing so will override the standard connectivity checks.`,
 		Run: hostannouncecmd,
 	}
+
+	hostFolderCmd = &cobra.Command{
+		Use:   "folder",
+		Short: "Add, remove, or resize a storage folder",
+		Long:  "Add, remove, or resize a storage folder.",
+	}
+
+	hostFolderAddCmd = &cobra.Command{
+		Use:   "add [path] [size]",
+		Short: "Add a storage folder to the host",
+		Long:  "Add a storage folder to the host, specifying how much data it should store",
+		Run:   wrap(hostfolderaddcmd),
+	}
+
+	hostFolderRemoveCmd = &cobra.Command{
+		Use:   "remove [path]",
+		Short: "Remove a storage folder from the host",
+		Long: `Remove a storage folder from the host. Note that this does not delete any
+data; it will instead be distributed across the remaining storage folders.`,
+
+		Run: wrap(hostfolderremovecmd),
+	}
+
+	hostFolderResizeCmd = &cobra.Command{
+		Use:   "resize [path] [size]",
+		Short: "Resize a storage folder",
+		Long: `Change how much data a storage folder should store. If the new size is less
+than what the folder is currently storing, data will be distributed across the
+other storage folders.`,
+		Run: wrap(hostfolderresizecmd),
+	}
+
+	hostSectorCmd = &cobra.Command{
+		Use:   "sector",
+		Short: "Add or delete a sector",
+		Long:  "Add or delete a sector. Note that deleting a sector may impact host revenue.",
+	}
+
+	hostSectorDeleteCmd = &cobra.Command{
+		Use:   "delete [root]",
+		Short: "Delete a sector",
+		Long: `Delete a sector, identified by its Merkle root. Note that deleting a
+sector may impact host revenue.`,
+		Run: wrap(hostsectordeletecmd),
+	}
 )
 
+// hostcmd is the handler for the command `siac host`.
+// Prints info about the host and its storage folders.
+func hostcmd() {
+	hg := new(api.HostGET)
+	err := getAPI("/host", &hg)
+	if err != nil {
+		die("Could not fetch host settings:", err)
+	}
+	sg := new(api.StorageGET)
+	err = getAPI("/storage", &sg)
+	if err != nil {
+		die("Could not fetch storage info:", err)
+	}
+
+	fm := hg.FinancialMetrics
+	is := hg.InternalSettings
+	nm := hg.NetworkMetrics
+
+	// calculate total storage available and remaining
+	var totalstorage, storageremaining uint64
+	for _, folder := range sg.StorageFolderMetadata {
+		totalstorage += folder.Capacity
+		storageremaining += folder.CapacityRemaining
+	}
+
+	// convert accepting bool
+	accept := yesNo(is.AcceptingContracts)
+	// convert price to SC/TB/mo
+	price, err := modules.StoragePriceToHuman(is.MinimumStoragePrice)
+	if err != nil {
+		price = ^uint64(0)
```

Comment:
There should be a comment here, or better yet an explicit error check. Right now, the only error that can be returned by the `StoragePriceToHuman` function is if there are more coins than can be represented by a uint64, but that's not obvious just by looking at this code.